Add fine-grained error view templates

JoeChapman opened the pull request:
So we can override the template served from the error middleware
based upon the particular http status code returned from the server.

Will need to amend the error middleware so that it serves the template based on the status code.
By adding these templates, we can ensure that when the middleware is updated there will be default templates to serve on those "errors"

JoeChapman committed:
Add fine-grained error view templates

So we can override the template served from the error middleware
based upon the particular http status code returned from the server.

lennym commented:
Any chance of having these all include a common template/partial rather than cloning the exact same html across several files?

sulthan-ahmed reviewed (approved):
LGTM

lennym commented:
Actually, the best way to do this is *I think* to just have the content of each file just be:

```
{{<error}}{{/error}}
```

Then an extension can:

```
{{<error}}
  {{$content}}
    ...stuff here...
  {{/content}}
{{/error}}
```

lennym committed:
"Try again" on general error instead of restarting

Uses template names rather than status codes for specific error cases. Make button on general error show try again by default and link to the current page.

Session expiry template overrides the default error to link to the start page, and show "start again".

JoeChapman commented on views/error.html:
```diff
@@ -4,7 +4,7 @@
   {{$content}}
     <p>{{{content.message}}}</p>
-    <a href="/" class="button" role="button">{{#t}}buttons.start-again{{/t}}</a>
+    <a href="" class="button" role="button">{{#t}}buttons.try-again{{/t}}</a>
```
will `href=""` reload the current page?

lennym replied:
Yes.
